### PR TITLE
initial 5.2 updates & Java version changes

### PR DESCRIPTION
**Pull request by hflynn**
See https://trello.com/c/esod316Q/75-document-dropping-java-1-6 and https://trello.com/c/wSW2lRBp/76-what-s-new-for-5-2-pages-version-history

Summarises 5.2 changes (details still to add for devs / OMERO.web), updates hardcoded version numbers and references to Java 1.6 as minimum requirement.

N.B. have made a separate card for version requirements page, won't tackle that on this PR.

--no-rebase

### CODE REVIEW

**Comment by hflynn** [comment]
Will be staged at https://www.openmicroscopy.org/site/support/omero5.2-staging/index.html

File: omero/developers/whatsnew.txt [hunk 1]
```diff
@@ -1,27 +1,7 @@
-What's new for OMERO 5.1 for developers
+What's new for OMERO 5.2 for developers
 =======================================
 
-- The :doc:`/sysadmins/version-requirements` section provides extensive
```
Comment (rleigh-dundee) [inline omero/developers/whatsnew.txt]: Isn't this initial point about the version requirements equally applicable to 5.2 and subsequent releases?  The following points were new 5.1 features and are fine to remove, but not sure about this one.

File: omero/sysadmins/whatsnew.txt [hunk 2]
```diff
@@ -1,88 +1,13 @@
-What's new for OMERO 5.1 for sysadmins
+What's new for OMERO 5.2 for sysadmins
 ======================================
 
-- The :doc:`version-requirements` section provides extensive details about
```
Comment (rleigh-dundee) [inline omero/sysadmins/whatsnew.txt]: As for the other section, why are the version requirements no longer needed?

**Comment by rleigh-dundee** [comment]
A few minor points added.  One major thing I think could be improved is that this deletes entirely the new features in 5.1 rather than adding a new section for the 5.2 features.  It would be nice if we instead added a `What's new in 5.2` as a new section at the top, leaving the `What's new in 5.1` section below it.  This keeps all the features we've added over time quite visible, and is also useful as a reminder for people upgrading from 5.0 or earlier.

File: omero/developers/whatsnew.txt [hunk 3]
```diff
@@ -47,51 +23,12 @@ The API's request operations :javadoc:`Chgrp
 <slice2html/omero/cmd/Chown.html>`, :javadoc:`Delete
 <slice2html/omero/cmd/Delete.html>`, and their superclass
 :javadoc:`GraphModify <slice2html/omero/cmd/GraphModify.html>`, are now
```
Comment (mtbc) [inline omero/developers/whatsnew.txt]: maybe not "are now deprecated" but "were deprecated in 5.1"?

Comment (mtbc) [inline omero/developers/whatsnew.txt]: "Chmod" and "Chmod2" should also be mentioned (similar URLs as for Chgrp) but awkwardly wasn't deprecated until 5.1.

**Comment by hflynn** [comment]
These pages weren't intended to replace the version history 3 times over, which is why I deleted the 5.1 features.
I did get rather delete happy removing the version requirements section though, I'll add that back in.

File: omero/users/clients-overview.txt [hunk 4]
```diff
@@ -16,10 +16,10 @@ installations worldwide, OMERO has been shown to be relatively easy to install
 and get running.
 
 OMERO.insight and OMERO.importer are desktop applications written in Java
-and require Java 1.6 (or higher) to be installed on the user's computer
+and require Java 1.7 (or higher) to be installed on the user's computer
 (automatically installed on most up-to-date OS X and Windows XP systems).
 
-Our **NEW** user assistance :help:`help website<>` provides a series of 
+Our user assistance :help:`help website<>` provides a series of
```
Comment (mtbc) [inline omero/users/clients-overview.txt]: `git grep -i 'new.*\b5\.1\b'` finds some "new in OMERO 5.1" kind of stuff, I don't know if any of that is also worth removing.

**Comment by mtbc** [comment]
You may wish to change `formats/conf.py` to update "omero_doc" to "/support/omero5.2/whatever". I don't know if it's worth a card about when to update the "5.1" in `omero/sysadmins/unix/walkthrough/README.md`, I'm not sure about that file.

File: omero/users/clients-overview.txt (hunk 4 above)
Comment (hflynn) [inline omero/users/clients-overview.txt]: I looked through 5.1 refs yesterday and didn't see anything that needed removing, although I will double-check with that command and maybe rephrase to 'introduced in 5.1' or similar.

**Comment by hflynn** [comment]
I'm leaving the sysadmin install docs to Simon - there's a separate card on the board for that stuff. I'll check the conf.py though, that had slipped my mind.

**Comment by hflynn** [comment]
Having thought some more, I'm going to make a card for the conf.py and hold off changing it for now.

**Comment by mtbc** [comment]
Happy to leave all these things to your judgment. :+1: Good with you too @rleigh-dundee?

**Comment by hflynn** [comment]
Latest commit addresses the stuff you grepped @mtbc

**Comment by hflynn** [comment]
Ok, this is just waiting on developer & OMERO.web updates, which may happen on new PR(s)

**Comment by rleigh-dundee** [comment]
@mtbc Yes, fine with me, thanks.

**Comment by hflynn** [comment]
Thanks both, probably easiest to merge this now and then add to the sysadmins & devs whats new files later rather than leaving this lying around to start conflicting. I'll amend cards accordingly.
